### ISSUE TITLE
Duplicate and missing sets in the cardset endpoint

### ISSUE THREAD

**Issue by Cubikon** [AlanOC91/YGOPRODeck#257]
In the cardinfo endpoint there is only a set name Yugi & Kaiba Collector Box and in the cardsets endpoint there is Yugi & Kaiba Collector Box, Yugi Collector Box and Kaiba Colelctor Box. I think you should remove the combined name and only keep the single names in the cardset endpoint and cardinfo endpoint, because the 2 cards in this sets have a different set code.

Missing sets in the cardsets endpoint:
- OTS Tournament Pack 15
- The Lost Art Promotion 2020 O
- The Lost Art Promotion 2020 K
- The Lost Art Promotion 2021 A
- The Lost Art Promotion 2021 B
- all OTS Tournament packs (POR) 
- Blazing Vortex Premiere! promotional card
- Duelist League Series 1 participation cards
-  Duelist League Series 2 participation cards
-  Duelist League Series 3 participation card
-  Duelist League Series 4 participation card
- JMP-EN005 Ultimate Edition
-Trials of the Pharaoh - Speed Duel: Battle City Box promotional card
- Trials of the Pharaoh - Match of the Millennium & Twisted Nightmares promotional card
- Trials of the Pharaoh - Trials of the Kingdom promotional card
- Summoned Skull Sample promotional card

Wrong set name in the cardinfo endpoint:
-WB01-EN001 and 002 super rare ( the "cards" in the set name needs to be "Cards")
- MP20-EN041,MP20-EN246,MP20-EN247,MP20-EN248 2 times in the list with set name "2020 Tin of Lost Memories" and "2020 Tin of Lost Memories Mega pack"
-  Yu-Gi-Oh! 5D's Tag Force 5 Promotional Cards ( in cardinfo endpoint with "cards" and "Cards", and in the cardsets endpoint only with "cards".

set in cardsets endpoint but there is none card in cardinfo endpoint with this set:
- Yu-Gi-Oh! Day April 2019 promotional card
- Speed Duel: Scars of Battle Launch Event participation card
- Yu-Gi-Oh! Championship Series 2019 prize card
- SJ Jump Pack Spring 2020 promotional card

I hope this can help

**Comment by AlanOC91** [comment]
> In the cardinfo endpoint there is only a set name Yugi & Kaiba Collector Box and in the cardsets endpoint there is Yugi & Kaiba Collector Box, Yugi Collector Box and Kaiba Colelctor Box. I think you should remove the combined name and only keep the single names in the cardset endpoint and cardinfo endpoint, because the 2 cards in this sets have a different set code.

I see the issue alright, I'll look into this.

> Missing sets in the cardsets endpoint

These should now be fixed!

> -WB01-EN001 and 002 super rare ( the "cards" in the set name needs to be "Cards")

Fixed!

> MP20-EN041,MP20-EN24...

Fixed!

> Yu-Gi-Oh! 5D's Tag Force 5....

Fixed!

> set in cardsets endpoint but there is none card in cardinfo endpoint with this set:

Hmmm I checked these and they seem to be working?

https://db.ygoprodeck.com/api/v7/cardsetsinfo.php?setcode=YDPR-ENS01

**Comment by Cubikon** [comment]
> Hmmm I checked these and they seem to be working?
> 
> https://db.ygoprodeck.com/api/v7/cardsetsinfo.php?setcode=YDPR-ENS01

Oh you are right. I overlooked them. i had a filter with only tcg cards.

The set name of the cards PCY-004 and PCY-005 are wrong. They were only in the Yu-Gi-Oh! Power of Chaos: Yugi the Destiny Limited Collector's Edition and not in the Yu-Gi-Oh! Power of Chaos: Yugi the Destiny promotional cards.

I also found over 500 cards that do not have a cardsets array.
I think they are all cards that are ocg-only. 



[missing_cardset_array.txt](https://github.com/AlanOC91/YGOPRODeck/files/6075639/missing_cardset_array.txt)
